### DISCUSSION TITLE
"watch" doesn't invoke callback

### DISCUSSION THREAD

**Discussion by LordUwe** [pmndrs/valtio#961] [Q&A]
When changing the `store.count` from within the timeout, the watch-callback is not invoked. The subscribe seems to work. Why is here a difference? Thanks!

```
import { proxy, subscribe } from 'valtio';
import { watch } from 'valtio/utils';

export const store = proxy({ count: 0 });

watch(get => {
  get(store);
  console.log('store.count from watch: ', store.count);
});

subscribe(store, () => {
  console.log('store.count from subscribe: ', store.count);
});

store.count = 2;

setTimeout(() => {
  store.count = 3;
  console.log('updated to 2');
}, 1000);
```

Output:
```
store.count from watch:  0
store.count from watch:  2
store.count from subscribe:  2
updated to 2
store.count from subscribe:  3
```

**Comment by dai-shi** [comment]
I tried to reproduce this.
https://stackblitz.com/edit/vitejs-vite-w9okvy?file=src%2FApp.tsx&terminal=dev

But, it works well:
<img width="392" alt="image" src="https://github.com/user-attachments/assets/9210ffdf-347d-4df2-82b1-f65db93fe572">

**Reply by LordUwe** [reply]
Thanks for checking. You are right. It works in the example. But somehow not in my environment/workspace. I'll try to find out why. Thanks again for your time.

**Reply by LordUwe** [reply]
I have an async function in another file used inside `watch`, where I fetch some data. This seems to cause the problem. It's similar to this:

```
const store = proxy({ count: 0 });

watch(async (get) => {
  get(store);
  // do async stuff here using the store as a parameter and change another store with the result value
  console.log('store.count from watch async: ', store.count);
});

watch((get) => {
  get(store);
  console.log('store.count from watch: ', store.count);
});

subscribe(store, () => {
  console.log('store.count from subscribe: ', store.count);
});

store.count = 2;

setTimeout(() => {
  store.count = 3;
  console.log('updated to 3');
}, 200);
```

It should be reproducible with the code above.

![image](https://github.com/user-attachments/assets/22f0884b-238f-41ea-9f76-e26970abd145)

**Reply by dai-shi** [reply]
Is this related? #924
